### PR TITLE
[script] enable TCP, DNS_CLIENT_OVER_TCP and OTBR_DHCP6_PD for 1.3 and 1.4 builds

### PR DESCRIPTION
**Pull request by suveshpratapa**
* `OTBR_BORDER_ROUTING_DHCP6_PD` has been renamed to `OTBR_DHCP6_PD`
* Update openthread, ot-br-posix and ot-nrf528xx to latest for 1.4 reference builds

### CODE REVIEW

**Review by LuDuda — APPROVED** [review]

File: script/otbr-setup.bash [hunk 1]
```diff
@@ -72,8 +72,11 @@ readonly OTBR_THREAD_1_3_OPTIONS=(
     "-DOT_BORDER_ROUTING=ON"
     "-DOT_SRP_CLIENT=ON"
     "-DOT_DNS_CLIENT=ON"
-    "-DOTBR_TREL=OFF"
-    "-DOTBR_NAT64=OFF"
+    "-DOT_TCP=ON"
```
Comment (sunytt) [inline script/otbr-setup.bash:75]: Do we want to enable these 1.4 features for 1.3 build?

Comment (jwhui) [reply script/otbr-setup.bash:75]: TREL, NAT64, and PD were all described in the Thread 1.3.0 specification.

@EskoDijk , any thoughts here?

Comment (sunytt) [reply script/otbr-setup.bash:75]: I see, so they are 1.3 features but in the 1.4 test plan. LGTM

**Review by sunytt — APPROVED** [review]

File: script/otbr-setup.bash (hunk 1 above)
Comment (EskoDijk) [reply script/otbr-setup.bash:75]: @jwhui Agree that any features defined in the 1.3.0 specification (even though not cert-tested yet for 1.3), should be enabled for a 1.3 build. And for a 1.4 build of course.  In terms of features enabled in the code, the really new thing in 1.4 is ePSKc support.

**Review by EskoDijk — APPROVED** [review]
Looks okay and as intended (as far as I can see!)

**Pull request merged**